netutils/ptpd/ptpd.c

```diff
@@ -1346,6 +1469,7 @@ static int ptp_daemon(int argc, FAR char** argv)
       ptp_periodic_send(state);
 
       state->selected_source_valid = is_selected_source_valid(state);
+      ptp_process_statusreq(state);
```

Comment:
why not process in the signal handler? it suffer the race condition process it here.

Reply:
If I understand correctly, signal handler could get invoked in middle of updating the fields in ptp_state_s. Would need masking & unmasking the signal then before write access to ptp_state_s. I also wanted to avoid any long processing in the signal handler, though that probably makes little difference.

It's true that processing it here could cause a race condition if two high-priority tasks call ptpd_status() simultaneously. Then one of the requests would timeout.

Reply:
> If I understand correctly, signal handler could get invoked in middle of updating the fields in ptp_state_s. Would need masking & unmasking the signal then before write access to ptp_state_s.

This behavior is done by signal subsystem by default.

> I also wanted to avoid any long processing in the signal handler, though that probably makes little difference.
> 

The signal handler copy a small struct, so the process should be very quickly.

> It's true that processing it here could cause a race condition if two high-priority tasks call ptpd_status() simultaneously. Then one of the requests would timeout.

Yes, it's also doesn't work in protected and kernel mode. So, it's better to use mqueue/socket, or creete a named share memory.

Reply:
I'm sorry, but I don't understand what kind of approach you are suggesting. `ptp_process_statusreq()` takes some time, and it also accesses multiple fields in `ptp_state_s`. If signal handler would occur in middle of e.g. `ptp_process_rx_packet()` and would then call `ptp_process_statusreq()` inside the signal handler, the fields could be in inconsistent state.

I think mqueue/socket is excessive for such simple status requests. Shared memory makes sense as future expansion, but for flat builds it is unnecessary.

If the task requesting status is lower priority than PTPD, simultaneous requests won't happen.

Reply:
> I'm sorry, but I don't understand what kind of approach you are suggesting. `ptp_process_statusreq()` takes some time, and it also accesses multiple fields in `ptp_state_s`.

ptp_process_statusreq just copy a bunch of small structure, plus sem_post, so the timing isn't a big issue.

> If signal handler would occur in middle of e.g. `ptp_process_rx_packet()` and would then call `ptp_process_statusreq()` inside the signal handler, the fields could be in inconsistent state.
> 

Yes, this is a big problem, it's a bad idea to call ptp_process_statusreq in signal handler.

> I think mqueue/socket is excessive for such simple status requests. Shared memory makes sense as future expansion, but for flat builds it is unnecessary.
> 
> If the task requesting status is lower priority than PTPD, simultaneous requests won't happen.

Ok, let's keep the current desgin.